From 6c4a67ece33d9551429490898bb3c793a689e913 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Thu, 24 Feb 2022 16:04:18 +0000
Subject: [PATCH] Improve detection of -fzero-call-used-regs=all support

GCC doesn't tell us whether this option is supported unless it runs into
the situation where it would need to emit corresponding code.
---
 m4/openssh.m4 | 3 +++
 1 file changed, 3 insertions(+)

--- a/m4/openssh.m4
+++ b/m4/openssh.m4
@@ -14,6 +14,8 @@ AC_DEFUN([OSSH_CHECK_CFLAG_COMPILE], [{
 	AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
 #include <stdlib.h>
 #include <stdio.h>
+/* Trivial function to help test for -fzero-call-used-regs */
+void f(int n) {}
 int main(int argc, char **argv) {
 	(void)argv;
 	/* Some math to catch -ftrapv problems in the toolchain */
@@ -21,6 +23,7 @@ int main(int argc, char **argv) {
 	float l = i * 2.1;
 	double m = l / 0.5;
 	long long int n = argc * 12345LL, o = 12345LL * (long long int)argc;
+	f(0);
 	printf("%d %d %d %f %f %lld %lld\n", i, j, k, l, m, n, o);
 	/*
 	 * Test fallthrough behaviour.  clang 10's -Wimplicit-fallthrough does
